source/common/upstream/cluster_manager_impl.cc

```diff
@@ -2399,6 +2399,20 @@ Http::ConnectionPool::InstancePtr ProdClusterManagerFactory::allocateConnPool(
   }
   if (protocols.size() == 1 && protocols[0] == Http::Protocol::Http2 &&
       context_.runtime().snapshot().featureEnabled("upstream.use_http2", 100)) {
+    if (host->cluster().clusterType().has_value() &&
+        host->cluster().clusterType()->name() == "envoy.clusters.reverse_connection") {
+      ENVOY_LOG_MISC(debug, "allocating reverse connection pool");
```

Comment:
If you really want to keep this log, letting ProdClusterManagerFactory inherit `protected Logger::Loggable<Logger::Id::pool>` could work.